k8sutils/pkg/cri/criwrapper.go

```diff
@@ -0,0 +1,142 @@
+package criwrapper
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	odigosv1 "github.com/odigos-io/odigos/api/odigos/v1alpha1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	criapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+type CriClient struct {
+	conn   *grpc.ClientConn
+	client criapi.RuntimeServiceClient
+	Logger logr.Logger
+}
+
+// Default runtime endpoints
+var defaultRuntimeEndpoints = []string{
+	"unix:///run/containerd/containerd.sock",
+	"unix:///run/crio/crio.sock",
+	"unix:///var/run/cri-dockerd.sock",
+}
+
+// Connect attempts to establish a connection to a CRI runtime.
+func (rc *CriClient) Connect() error {
+	var err error
+
+	for _, endpoint := range defaultRuntimeEndpoints {
+		rc.Logger.Info("Attempting to connect to CRI runtime", "endpoint", endpoint)
+
+		rc.conn, err = grpc.NewClient(
```

Comment:
Are you sure this is actually trying to connect and not just creating a client?
Few thoughts:
- As only one of the default Endpoints will respond - maybe worth putting a timeout when opening a connection as some of them will just time out so we can bail quickly
- I think the CRI implementation is part of the container id in Kubernetes maybe we can use that (for example getting the container id of the current Odiglet Pod) and connector only to the right one instead of trying all the options

Reply:
1. I Warp it up as a Connect() -  The function’s primary purpose is to connect to the CRI runtime, which it achieves by establishing and validating a gRPC connection.

I change the logic so it will try to find the unix domain socket in the FS so we wont need to iterate over them and try to connect to each one of them